R/estimate_time_varying.R

```diff
@@ -80,69 +80,74 @@ estimate_time_varying <- function(df_in,
   # TODO input checking
   checkmate::assert_logical(smooth_inputs, len = 1L, any.missing = FALSE)
   checkmate::assert_logical(correct_for_delays, len = 1L, any.missing = FALSE)
-  checkmate::assert_integerish(burn_in_value, lower = 0, len = 1L)
-  checkmate::assert_data_frame(df_in)
+  checkmate::assert_integerish(burn_in_value, lower = 1, len = 1L)
+  checkmate::assert_data_frame(data)
   checkmate::assert_logical(correct_for_delays, len = 1L)
+  checkmate::assert_integerish(smoothing_window, lower = 1, len = 1L)
 
   # returns error message if no delay distribution is supplied, but correction
   # for delays was requested
-  if (missing(epi_dist) && (isTRUE(correct_for_delays))) {
+  if (correct_for_delays && missing(epi_dist)) {
     stop(
-      "To correct for the delay between case detection and death,\
-       please provide an onset-to-death (or similar) `epidist` object"
+      "`epidist` object required when `correct_for_delays` is TRUE"
     )
   }
   if (!missing(epi_dist)) {
```

Comment:
epi_dist can't be missing because no default is provided

Reply:
Fixed in https://github.com/epiverse-trace/datadelay/pull/44/commits/809c4557a1949bb4cc13f4516597076e0343f919 and others

Reply:
The function would still run if `correct_for_delays` is false though. I agree the intention is to correct for delays, so would you prefer the `epi_dist` argument to have no default then?

Reply:
Yes, you're right, I missed that `epi_dist` is only used when `correct_for_delays` is true so it's probably fine to have a default there.